### ISSUE TITLE
[Discussion] Monitoring within the snapshotter

### ISSUE THREAD

**Issue by louisportay** [gardener/etcd-backup-restore#544]
**Feature:**

As the snapshotter already gathers metrics, could the idea of exposing them with a HTTP server be realistic ?
I know it's feasible but is it the goal of the snapshotter ? I think it is a really nice to have, not to hard to implement.

**Motivation (Why is this needed?):**

I tried using the command server in my environment but I found it hard to do exactly what I wanted (just export snapshotter metrics).

**Approach/Hint to the implement solution:**

start HTTP server with /metrics route just like the server command does.

If this feature gets approval, I can draft an implementation

**Comment by ishan16696** [comment]
I didn't get the motivation behind this feature requirement. Isn't all the metrics including (snapshotter related metrics) can be retrieved using the endpoint `/metrics`  ?
How it will be different from the existing snapshotter metrics ? do you want to add more metrics ?

**Comment by louisportay** [comment]
Yes, metrics can be retrieved with the server command, but the server command is doing a lot more things than what I need (the snapshotter does a perfect job for me) but I have no metrics related to the snapshotters.
If I read the code right, the only command exposing an HTTP server is _server_, I propose to add one to the _snapshotter_ command (as it's a long running process as well).

**Comment by ishan16696** [comment]
so, you want to start separate HTTP server for `snapshotter` sub-command to retrieve the `snapshotter` related metrics right ? 
TBH, I don't find any good reason to add this as a new feature. It will increase the code complexity with very less value added.
I would suggest to use the backup-restore with server command, it will you give much more functionalities like restoration, etcd data-dir validation, snapshotter and defragmentation etc.
But If you want still want to use only `snapshotter` then I would suggest what you can do is try to disable other functionality. 
To disable defragmentation use this flag  `--defragmentation-schedule`.
To disable the restoration don't use our [etcd custom image](https://github.com/gardener/etcd-custom-image), you can use normal etcd image, so that there will be no [initialization](https://github.com/gardener/etcd-custom-image/blob/master/etcd_bootstrap_script.sh#L46-L50) requests.
Use this flag `--enable-member-lease-renewal`,`--enable-snapshot-lease-renewal` to disable member lease renewal.
Hope it works.

**Comment by ishan16696** [comment]
cc @unmarshall @aaronfern 
/need-second-opinion

**Comment by ishan16696** [comment]
Hi @louisportay,
may be I have misunderstood your requirements, let me rephrase:
Do you want to add a separate endpoint exposed via HTTP server which only contains `snapshotter` metrics ?
or Do you want to start separate HTTP server for `snapshotter` sub-command to retrieve the snapshotter related metrics?

**Comment by louisportay** [comment]
Hi Ishan,

The latter one, a separate HTTP server packed with `snapshotter` sub-command, I think the implementation should be straightforward and similar to the `server` sub-command one.
I think there should be an easy way to retrieve these metrics without the added overhead of tuning configuration/editing code for the `server` sub-command.

What do you think ?

**Comment by louisportay** [comment]
Welp, I guess the discussion is over.

Thanks for your time gentlemen
